jupyter-web-app/rockcraft.yaml

```diff
@@ -1,88 +1,110 @@
 
-      # build backend kubeflow library
-      cp $KF_REPO/src/components/crud-web-apps/common/frontend/kubeflow-common-lib/package.json ./
-      cp $KF_REPO/src/components/crud-web-apps/common/frontend/kubeflow-common-lib/package-lock.json ./
+  frontend-lib:
+    plugin: nil
+    source: https://github.com/kubeflow/kubeflow
+    source-type: git
+    source-tag: v1.7-branch  # upstream branch
+    source-depth: 1
+    build-snaps:
+      - node/12/stable
+    build-environment:
+      - NG_CLI_ANALYTICS: "ci"
+    override-build: |
+      cd components/crud-web-apps/common/frontend/kubeflow-common-lib
       npm ci
-
-      cp $KF_REPO/src/components/crud-web-apps/common/frontend/kubeflow-common-lib/projects ./ -r
-      cp $KF_REPO/src/components/crud-web-apps/common/frontend/kubeflow-common-lib/angular.json .
-      cp $KF_REPO/src/components/crud-web-apps/common/frontend/kubeflow-common-lib/tsconfig.json .
       npm run build
+      cp -r dist/kubeflow/ $CRAFT_STAGE
 
-      # build frontend
-      cp $KF_REPO/src/components/crud-web-apps/jupyter/frontend/package.json ./
-      cp $KF_REPO/src/components/crud-web-apps/jupyter/frontend/package-lock.json ./
-      cp $KF_REPO/src/components/crud-web-apps/jupyter/frontend/tsconfig.json ./
-      cp $KF_REPO/src/components/crud-web-apps/jupyter/frontend/tsconfig.app.json ./
-      cp $KF_REPO/src/components/crud-web-apps/jupyter/frontend/tsconfig.spec.json ./
-      cp $KF_REPO/src/components/crud-web-apps/jupyter/frontend/angular.json ./
-      cp $KF_REPO/src/components/crud-web-apps/jupyter/frontend/src ./ -r
+  frontend:
+    after: [frontend-lib]
+    plugin: nil
+    source: https://github.com/kubeflow/kubeflow
+    source-type: git
+    source-tag: v1.7-branch  # upstream branch
+    source-depth: 1
+    build-snaps:
+      - node/12/stable
+    build-environment:
+      - NG_CLI_ANALYTICS: "ci"
+    override-build: |
+      cd components/crud-web-apps/jupyter/frontend
       npm ci
-
-      # install jupyter ui web app
-      cp -r ./dist/kubeflow/ ./node_modules/kubeflow
+      cp -r $CRAFT_STAGE/kubeflow/ ./node_modules/  # TODO confirm
       npm run build -- --output-path=./dist/default --configuration=production
-      pip3 install .
+      cp -r dist/default $CRAFT_STAGE
 
-      cp $KF_REPO/src/components/crud-web-apps/jupyter/backend/requirements.txt .
-      pip3 install -r requirements.txt
+  webapp:
+    after: [backend, frontend]
+    plugin: nil
+    source: https://github.com/kubeflow/kubeflow
+    source-type: git
+    source-tag: v1.7-branch  # upstream branch
+    source-depth: 1
+    build-packages:
+      - python3-venv
+      - python3-setuptools
+      - python3-pip
+    override-build: |
+      pip3 install $CRAFT_STAGE/kubeflow-1.1-py3-none-any.whl
+      cd components/crud-web-apps/jupyter/backend
+      cp -r $CRAFT_STAGE/default apps/default/static/
+      cp -r apps $CRAFT_STAGE/
+      cp entrypoint.py $CRAFT_STAGE
```

Comment:
can't you copy this directly to `$CRAFT_PART_INSTALL` instead of duplicating the command below?

Reply:
Updated.